src/ixdat/db.py

```diff
@@ -260,7 +323,61 @@ def __init__(self, i, cls):
         """
         self.id = i
         self.cls = cls
+        if not backend or backend == "none" or backend is database_backends["none"]:
+            raise DataBaseError(
+                f"Can't make a PlaceHolderObject with backend={backend}"
+            )
+        self.backend = backend
 
     def get_object(self):
         """Return the loaded real object represented by the PlaceHolderObject"""
-        return self.cls.get(self.id)
+        return self.cls.get(self.id, backend=self.backend)
+
+    @property
+    def identity(self):
+        """Placeholder also has an identity to check equivalence without loading"""
+        if self.backend is DB.backend:
+            return self.id
+        return self.backend, self.id
+
+
+def fill_object_list(object_list, obj_ids, cls=None):
+    """Add PlaceHolderObjects to object_list for any unrepresented obj_ids.
+
+    Args:
+        object_list (list of objects or None): The objects already known,
+            in a list. This is the list to be appended to. If None, an empty
+            list will be appended to.
+        obj_ids (list of ints or None): The id's of objects to ensure are in
+            the list. Any id in obj_ids not already represented in object_list
+            is added to the list as a PlaceHolderObject
+        cls (Saveable class): the class remembered by any PlaceHolderObjects
+            added to the object_list, so that eventually the right object will
+            be loaded.
+    """
+    cls = cls or object_list[0].__class__
```

Comment:
We just said above that object_list might be None, so there may need to be a check and an exception here, if both it and cls is None

Reply:
I just added to the docstring that cls must be specified if object_list is empty.